### PR TITLE
Feature/vector math

### PR DESCRIPTION
**Pull request by JorisWillig**

### CODE REVIEW

**Comment by notzain** [comment]
Unit test

File: engine/common/Vector2D.h [hunk 1]
```diff
@@ -97,6 +99,29 @@ struct Vector2D {
...
+
+    friend bool operator==(const Vector2D left, const Vector2D& right) {
+        return left.x == right.x && left.y == right.y;
+    }
+    friend bool operator!=(const Vector2D left, const Vector2D right) {
+        return left.x != right.x || left.y != right.y;
+    }
+
+    // mathematical properties
     Vector2D max(Vector2D other)
```
Comment (notzain) [inline engine/common/Vector2D.h]: Both min, max ->

Wouldn't a static function make more sense?

Vector2D::max(left, right);

File: engine/common/Vector2D.h [hunk 2]
```diff
@@ -22,6 +23,7 @@ struct Vector2D {
         return static_cast<int>(y);
```
Comment (notzain) [inline engine/common/Vector2D.h]: Should Vector2D care about this? 

The caller has direct access to x and y. If they need it as an int, they should probably do it themselves.

Comment (FabioGNR) [reply engine/common/Vector2D.h]: Makes sense but it lead to a lot of casts. For a rectangle there would be 4 casts which lead to, in my opinion, unreadable/ugly code

Comment (JorisWillig) [reply engine/common/Vector2D.h]: I agree with Fabio. Zain has a good point but in this case I think the cons of putting it in the struct (i.e. going against convention) don't outway the pros.

**Comment by notzain** [comment]
Reminder:

Check where to put Vector2D, including namespaces.

File: game/systems/RenderSystem.cpp [hunk 3]
```diff
@@ -14,8 +14,8 @@ namespace systems {
             auto& positionComponent = world.getComponent<components::PositionComponent>(entity);
             auto& dimensionComponent = world.getComponent<components::DimensionComponent>(entity);
 
-            common::Vector2D position{ positionComponent.position.x * m_unitSize, positionComponent.position.y * m_unitSize };
-            common::Vector2D size{ dimensionComponent.dimension.x * m_unitSize, dimensionComponent.dimension.y * m_unitSize };
+            common::Vector2D<int> position{ (int)positionComponent.position.x * m_unitSize, (int)positionComponent.position.y * m_unitSize };
```
Comment (FabioGNR) [inline game/systems/RenderSystem.cpp]: static_cast?

**Pull request merged**